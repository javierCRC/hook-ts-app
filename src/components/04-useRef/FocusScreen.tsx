import React, { MutableRefObject, useRef } from 'react'

import './effects.css';

export const FocusScreen = () => {
     
   
    const vInputRef: any = useRef();
      //console.log( vInputRef );

    const f_handleClick = () => {

        vInputRef.current.select();

    }

    return (
        <div>
            <h1> Focus Screen with useRef </h1>
            <hr />

            <input ref = { vInputRef }
                   className='form-control'
                   placeholder='Put your name'
            
            />

            <button 
                 className='btn btn-outline-primary m-3'
                 onClick={ f_handleClick }
            >
                Focus <i className="fas fa-eye"></i>
                
            </button>
            
        </div>
    )
}
