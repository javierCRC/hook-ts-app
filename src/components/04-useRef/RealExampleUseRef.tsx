import React,{ useState } from 'react'

import { MultipleCustomHook } from '../03-Examples/MultipleCustomHook';

import './effects.css';

export const RealExampleUseRef = () => {
   
    
    // use state to show or hide component
    const [show, setShow] = useState(false);
   
    const f_handleClickShow = () => {
       setShow( !show );
    }

    return (
        <div>
            <h1> Real Example of useRef in TypeScript </h1>
            <hr />

            { show && <MultipleCustomHook /> }
            
            <button className="btn btn-outline-success"
                    onClick={ f_handleClickShow }
            >
                Show <i className="fas fa-eye"></i> / Hide <i className="fas fa-eye-slash"></i>
            </button>

            

        </div>
    )


}
