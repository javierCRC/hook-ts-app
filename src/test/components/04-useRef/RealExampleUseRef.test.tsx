import { renderHook,act } from '@testing-library/react-hooks';
import { RealExampleUseRef } from '../../../components/04-useRef/RealExampleUseRef';
import { shallow } from "enzyme"


describe('Test in component <RealExampleUseRef />', () => {

    
    
    test('1.Must do render the component RealExampleUseRef.', () => {
        
        
        // 1 Arrange *******************************************************************************
             
              let vExistComponent: boolean;
          
        //2. Act *******************************************************************************

             const cWrapper = shallow(<RealExampleUseRef />);
             
             vExistComponent = cWrapper.find('MultipleCustomHook').exists();

        //3. Assert *******************************************************************************

            expect(cWrapper).toMatchSnapshot();
            expect(vExistComponent).toBe(false); // the  component dont be exists yet

            //console.log( cWrapper.html() );
            
    });

   test('2.should be render the component MultipleCustomHook simulating click.', () => {
        
        
        // 1 Arrange *******************************************************************************
             
             let vExistComponent2: boolean;
          
        //2. Act *******************************************************************************

             const cWrapper = shallow(<RealExampleUseRef />);

             cWrapper.find('button').simulate('click');

             vExistComponent2 = cWrapper.find('MultipleCustomHook').exists();

        //3. Assert *******************************************************************************

            
            expect(vExistComponent2).toBe(true); // the  component dont be exists yet
       
    });

    
});
