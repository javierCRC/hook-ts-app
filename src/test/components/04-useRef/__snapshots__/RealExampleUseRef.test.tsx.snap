// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component <RealExampleUseRef /> 1.Must do render the component RealExampleUseRef. 1`] = `
<div>
  <h1>
     Real Example of useRef in TypeScript 
  </h1>
  <hr />
  <button
    className="btn btn-outline-success"
    onClick={[Function]}
  >
    Show 
    <i
      className="fas fa-eye"
    />
     / Hide 
    <i
      className="fas fa-eye-slash"
    />
  </button>
</div>
`;
